src/webserver/HttpMessage.java

```
@@ -0,0 +1,214 @@
+   * retrieved on a call to {@link #getHeader(String)} with the appropriate header
+   * name.
+   * <p>
+   * This constructor will throw out any headers that do not adhere to this rule.
+   * 
+   * @param headers a {@code String} array with applicable headers, with key and
+   *                value seperated by {@code :} for each header string.
+   */
+  public HttpMessage(String[] headers, String body) {
+    this.headers = new HashMap<>();
+    this.addHeaders(headers);
+    this.body = "";
+  }
+
+  /**
+   * Adds a header and value to this message's list of headers.
+   * 
+   * @param header  the name of the header to be added.
+   * @param details the details of the header.
+   */
+  public void addHeader(String header, String details) {
+    this.headers.put(header, details);
+  }
+
+  /**
+   * Adds a group of headers to this message's list of headers.
+   * 
+   * @param headers a header-name header-detail map of headers to add.
+   */
+  public void addHeaders(Map<String, String> headers) {
+    this.headers.putAll(headers);
+  }
+
+  /**
+   * Adds a string array of headers to this message's list of headers. Headers in
+   * the array will be parsed, seperated by a colon.
+   * <p>
+   * For example: {@code ["Connection: Keep-Alive", "Accept-Language: en-us"]}
+   * will be parsed into
+   * {@code Connection: "Keep-Alive", Accept-Language: "en-us"}.
+   * <p>
+   * Invalid headers will be thrown out.
+   * 
+   * @param headers a string array of headers.
+   */
+  public void addHeaders(String[] headers) {
+    for (String s : headers) {
+      // seperate the headers info and place into hashmap
+      if (s.indexOf(":") != -1) {
+        String headerName = s.substring(0, s.indexOf(":")).trim();
+        String headerDetails = s.substring(s.indexOf(":")).trim();
+
+        if (!headerName.equals("") && !headerDetails.equals("")) {
+          this.headers.put(headerName, headerDetails);
+        }
+      }
```

Maybe have a `void addHeader(String header)` too. In addition, don't silently ignore things like missing colon or empty key/value. The 400 response code exists for a reason. Abort here, then somewhere in the server can handle the exception with a 400 reply.

---

What would `void addHeader(String header)` do? Would it assign some default value to the header, or do some headers support no value inside them?

---

It splits a single String and adds it to the headers, as opposed to a method that can only take multiple strings.